db/migrate/20180403144853_convert_vm_attrs_to_hash.rb

```diff
@@ -0,0 +1,67 @@
+class ConvertVmAttrsToHash < ActiveRecord::Migration[5.1]
+  def up
+    transform :to_h
+  end
+
+  YML_HASH = '!ruby/hash:ActiveSupport::HashWithIndifferentAccess'
+  YML_PARAMS = /\!ruby\/[\w-]+:ActionController::Parameters/
+
+  def transform(transform_method)
+    say "Starting serialized attributes conversion, this can take long time based on data amount"
+    say "Converting Nics, total: #{FakeNic.unscoped.count}"
+    transform_batch_columns(FakeNic, [:attrs, :compute_attributes], transform_method)
+
+    say "Converting Compute Attributes, total: #{FakeComputeAttribute.unscoped.count}"
+    transform_batch_columns(FakeComputeAttribute, [:vm_attrs], transform_method)
+
+    say "Converting Compute Resources, total #{FakeComputeResource.unscoped.count}"
+    transform_batch_columns(FakeComputeResource, [:attrs], transform_method)
+
+    say "Converting Lookup Keys, total: #{FakeLookupKey.unscoped.count}"
+    transform_batch_columns(FakeLookupKey, [:default_value], transform_method)
+
+    say "Converting Lookup Values, total: #{FakeLookupValue.unscoped.count}"
+    transform_batch_columns(FakeLookupValue, [:value], transform_method)
+
+    say "All conversions finished"
+  end
+
+  def transform_batch_columns(base, serialized_columns, transform_method)
+    base.unscoped.select(serialized_columns + [:id]).find_each  do |object|
+      serialized_columns.each do |column|
+        attributes = object.send :read_attribute_before_type_cast, column
+        next if attributes.nil?
+        new_attributes = send(transform_method, attributes)
```

Comment:
since we removed the down version, we can get rid of `transform_method` everywhere.
Then, you can leverage the fact that `gsub!` returns `nil` if no changes are made to save the comparison below. so in total:
```ruby
attributes = object.send :read_attribute_before_type_cast, column
next if attributes.nil?
object.update_column(column, attributes) unless attributes.gsub!(YML_PARAMS, YML_HASH).nil?
```